gusty/__init__.py

```diff
@@ -236,3 +239,39 @@ def build_dag(directory, dag, latest_only=True):
     set_dependencies(yaml_specs, tasks, dag=dag, latest_only=latest_only)
 
     return tasks
+
+###############################################
+################## GustyDAG ###################
+###############################################
+
+class GustyDAG(airflow.DAG):
```

Comment:
It looks like the class here is not adding new behavior to Dag, but constructing it in a new way. I wonder if a factory function might be a good alternative?

Two potential limitations with GustyDag here are (1) it's not Liskov substitutable since its first arg in directory, not filename, and (2) it forces a pattern of subclassing in order to create a Dag object in new ways.

Here's an example that uses the same code, but in a factory function...

```python
def create_dag_from_yaml(directory, latest_only = True, **kwargs):
    # could wrap all this in a function ----
    name = os.path.basename(directory)

    metadata_file = os.path.join(directory, "METADATA.yml")
    if os.path.exists(metadata_file):
        with open(metadata_file) as inf:
            dag_metadata = yaml.load(inf, GustyYAMLLoader)

            # The keyword arguments take precedence over metadata,
            # except that default_args is also combined
            default_args = dag_metadata.get("default_args", {})
            default_args.update(kwargs.get("default_args", {}))

            dag_metadata.update(kwargs)
            kwargs = dag_metadata
            kwargs["default_args"] = default_args
    #  /wrapping ----
    
    # Initialize the DAG
    dag = GustyDag(name, **kwargs)

    yaml_specs = get_yaml_specs(directory)
    tasks = build_tasks(yaml_specs, dag=dag)
    set_dependencies(yaml_specs, tasks, dag=dag, latest_only=latest_only)
```

If you're wanting to extend with a lot of different ways of loading things, could implement as a class like (Note this is probably overkill) ...

```python
class DagConstructor:
    dag_klass = airflow.Dag

    @classmethod
    def from_yaml(cls):
        # same as factory function, but swap out Dag with cls.dag_klass
```